### PR TITLE
Add options to listunspent rpc call

### PR DESCRIPTION
**Pull request by pedrobranco**
Add options to `listunspent` rpc call:
- Remove address information
- Remove **0 value** UTXO's

Increases **listunspent** performance.

@promag @jpdffonseca

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -2383,11 +2388,23 @@ UniValue listunspent(const UniValue& params, bool fHelp)
         }
     }
 
+    bool fWithAddressInfo = true;
+    bool fIncludeZeroValue = true;
+    if (params.size() > 3) {
```
Comment (promag) [inline src/wallet/rpcwallet.cpp]: Add newline.

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -2407,10 +2424,12 @@ UniValue listunspent(const UniValue& params, bool fHelp)
         entry.push_back(Pair("txid", out.tx->GetHash().GetHex()));
         entry.push_back(Pair("vout", out.i));
         CTxDestination address;
-        if (ExtractDestination(out.tx->vout[out.i].scriptPubKey, address)) {
-            entry.push_back(Pair("address", CBitcoinAddress(address).ToString()));
-            if (pwalletMain->mapAddressBook.count(address))
-                entry.push_back(Pair("account", pwalletMain->mapAddressBook[address].name));
+        if (fWithAddressInfo){
```
Comment (promag) [inline src/wallet/rpcwallet.cpp]: Space before `{`.

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -2383,11 +2388,23 @@ UniValue listunspent(const UniValue& params, bool fHelp)
         }
     }
 
+    bool fWithAddressInfo = true;
+    bool fIncludeZeroValue = true;
+    if (params.size() > 3) {
+        const UniValue& options = params[3].get_obj();
+        if (options.exists("withaddressinfo")){
```
Comment (promag) [inline src/wallet/rpcwallet.cpp]: Newlines?

Comment (promag) [inline src/wallet/rpcwallet.cpp]: Space before `{`.

Comment (pedrobranco) [inline src/wallet/rpcwallet.cpp]: Newlines before any code block/if?

**Comment by pedrobranco** [comment]
@promag Maybe we should deprecate this?

**Comment by pedrobranco** [comment]
Closing as does not contribute that much for bitcoin's listunspent performance.

**Pull request closed without merging**